src/python/pants/backend/helm/target_types.py

```diff
@@ -489,6 +525,7 @@ class HelmDeploymentTarget(Target):
     alias = "helm_deployment"
     core_fields = (
         *COMMON_TARGET_FIELDS,
+        HelmDeploymentChartField,
         HelmDeploymentReleaseNameField,
         HelmDeploymentDependenciesField,
         HelmDeploymentSourcesField,
```

Comment:
Does having a `helm_deployment` and a `helm_chart` in the same directory work? Both of their sources fields pull in the same files, so they're owned by more than one target, but the files are the same so merging the digest should work.

Reply:
It could work if the user specifies the `sources` field for each target in a way that they don't overlap.

However the recommended approach is to have them in separate folders. It's also good practice to do so.